From f63c475d8b1de34d132bd5695b3ecf665be9868b Mon Sep 17 00:00:00 2001
From: Patrice Vilchez <patrice.vilchez@atmel.com>
Date: Wed, 16 Jan 2013 17:01:52 +0100
Subject: [PATCH 297/319] drivers/video/atmel_hlcdfb.c: fix bad timing
 configuration

The right correspondance between LCD margins and LCD timings is:
* upper margin -> vertical back porch
* lower margin -> vertical front porch
* left  margin -> horizontal back porch
* right margin -> horizontal front porch

 Signed-off-by: Patrice Vilchez <patrice.vilchez@atmel.com>
---
 drivers/video/atmel_hlcdfb.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/video/atmel_hlcdfb.c b/drivers/video/atmel_hlcdfb.c
index d1136a5..dd9dcee 100644
--- a/drivers/video/atmel_hlcdfb.c
+++ b/drivers/video/atmel_hlcdfb.c
@@ -297,13 +297,13 @@ static int atmel_hlcdfb_setup_core_base(struct fb_info *info)
 	dev_dbg(info->device, "  * LCDC_LCDCFG1 = %08lx\n", value);
 	lcdc_writel(sinfo, ATMEL_LCDC_LCDCFG1, value);
 
-	value = (info->var.lower_margin) << LCDC_LCDCFG2_VBPW_OFFSET;
-	value |= (info->var.upper_margin - 1) << LCDC_LCDCFG2_VFPW_OFFSET;
+	value = (info->var.upper_margin) << LCDC_LCDCFG2_VBPW_OFFSET;
+	value |= (info->var.lower_margin - 1) << LCDC_LCDCFG2_VFPW_OFFSET;
 	dev_dbg(info->device, "  * LCDC_LCDCFG2 = %08lx\n", value);
 	lcdc_writel(sinfo, ATMEL_LCDC_LCDCFG2, value);
 
-	value = (info->var.right_margin - 1) << LCDC_LCDCFG3_HBPW_OFFSET;
-	value |= (info->var.left_margin - 1) << LCDC_LCDCFG3_HFPW_OFFSET;
+	value = (info->var.left_margin - 1) << LCDC_LCDCFG3_HBPW_OFFSET;
+	value |= (info->var.right_margin - 1) << LCDC_LCDCFG3_HFPW_OFFSET;
 	dev_dbg(info->device, "  * LCDC_LCDCFG3 = %08lx\n", value);
 	lcdc_writel(sinfo, ATMEL_LCDC_LCDCFG3, value);
 
-- 
1.7.10.4

